posts/heatmaps-clustering/index.qmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Heatmaps done right"
+subtitle: "A note on clustering distance matrices before plotting. Think sns.clustermap using altair."
+date: "08/03/2022"
+image: "visualization.png"
+categories: [visualisation, how-to]
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .qmd
+      format_name: quarto
+      format_version: '1.0'
+      jupytext_version: 1.13.7
+  kernelspec:
+    display_name: Python 3.8.12 ('heatmap')
+    language: python
+    name: python3
+---
+
+Given a distance matrix, i.e. a symmetric matrix of distances between observations, if the indices of the observations are arbitrary, or more generally there is no variable by which we want to order the observations, then plotting a heatmap using the default ordering (of the indices) is often not very useful. A better approach is to cluster the observations, and use the new ordering. This is what `sns.clustermap` offers, see [docs](https://seaborn.pydata.org/generated/seaborn.clustermap.html).
+
+::: {.column-margin}
+_Question:_ Is there another/better way to re-order the indices? I don't think anything purely algebraic would work, as the before and after objects are fundamentally different matrices.
+:::
+
+We will use the USA presidential speeches to walk through the clustering, and then do the plotting using altair for interactivity - as much as I like seaborn (and it's [next generation API](https://seaborn.pydata.org/nextgen/) looks very cool), interactivity is great. Then again from the src, it looks like a lot of thought has been put into `clustermap`, so there might be other reasons to use it.
+
+This is therefore a trivial post, but I often find myself wanting to do this. Indeed, I will probably refer back to it myself. The data is chosen arbitrarily, and the reordering doesn't seem to add much in this case, though one could probably have fun checking political party, etc.
```

Comment:
Super minor: "This post is somewhat trivial but I often find myself wanting to do this, thus I expect to refer back to this post for reference myself" or something a bit simpler? As is now had made me stop at this sentence twice in two distinct occasions. Think maybe because it reads a bit formal and complex? Anyway, not fussed.

Reply:
In fact, given the next sentence is about data, how about the above point is made as a footnote?